firebase-vscode/src/data-connect/service.ts

```diff
@@ -243,23 +234,23 @@ export class DataConnectService {
       extensions: this._auth(),
     });
     if (params.instance === InstanceType.PRODUCTION) {
-      const resp = await executeGraphQL(servicePath, prodBody);
+      const client = dataconnectDataplaneClient();
+      const resp = await executeGraphQL(client, servicePath, prodBody);
       return this.handleProdResponse(resp);
     } else {
-      const resp = await fetch(
-        (await this.emulatorsController.getLocalEndpoint()) +
-          `/v1beta/${servicePath}:executeGraphql`,
-        {
-          method: "POST",
-          headers: {
-            Accept: "application/json",
-            "Content-Type": "application/json",
-            "x-mantle-admin": "all",
-          },
-          body,
-        },
-      );
-      return this.handleResponse(resp);
+      const endpoint = this.emulatorsController.getLocalEndpoint();
+      if (!endpoint) {
+        throw new DataConnectError(
+          `Emulator isn't running. Please start your emulator!`,
+        );
+      }
+      const client = new Client({
+        urlPrefix: endpoint,
+        apiVersion: "v1beta",
+        auth: true,
```

Comment:
Please revert this to use `"x-mantle-admin": "all"` instead. `auth: true` may send production auth tokens to the emulator and may run into issues when developing offline.

If the `Client` API doesn't do what we want, then using `fetch` is also okay for now. We can always clean code up later but let's not mess with the behavior.

Reply:
What about remove `auth: true` here?

`x-mantle-admin` doesn't matter.

Reply:
I tested executing against emulator. Everything works fine, both error and success.

Reply:
Here are some emulator logs: https://paste.googleplex.com/6332029630414848

Good catch on not sending auth token. Emulator has V(3) debug logs for the auth token, won't log by default even if someone sends a real token to it.